app/src/main/java/io/appium/uiautomator2/model/UiObjectElement.java

```diff
@@ -114,19 +116,48 @@ public boolean getBoolAttribute(final String attr)
 
     public void setText(final String text, boolean unicodeKeyboard) throws UiObjectNotFoundException {
         String textToSend = text;
+        AccessibilityNodeInfo nodeInfo = AccessibilityNodeInfoGetter.fromUiObject(element);
```

Comment:
Can we combine this to avoid unnecessary code duplication? for example by creating a static helper method